crypto/bls.go

```diff
@@ -283,9 +330,15 @@ func (a *blsBLS12381Algo) decodePublicKey(publicKeyBytes []byte) (PublicKey, err
 	if err != nil {
 		return nil, fmt.Errorf("decode public key failed %w", err)
 	}
-	if !pk.point.checkValidPublicKeyPoint() {
+
+	// membership check in G2
+	if C.check_membership_G2((*C.ep2_st)(&pk.point)) != valid {
 		return nil, invalidInputsErrorf("input key is infinity or does not encode a BLS12-381 point in the valid group")
 	}
+
+	// check point is non-infinity and cache it
+	pk.isIdentity = C.ep2_is_infty((*C.ep2_st)(&pk.point)) != valid
```

Comment:
Same comment here, for making a wrapper function for this.

Reply:
https://github.com/onflow/flow-go/pull/3421/commits/4f3197b9e3843ed4b693744fff1bd4b90174cdbc